src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -4997,7 +4997,7 @@ void AuraEffect::HandleAuraDummy(AuraApplication const* aurApp, uint8 mode, bool
         // AT REMOVE
         else
         {
-            if ((GetSpellInfo()->IsQuestTame()) && caster && caster->IsAlive() && target->IsAlive())
+            if ((GetSpellInfo()->IsQuestTame()) && caster && caster->IsAlive() && target->IsAlive() && aurApp->GetRemoveMode() != AURA_REMOVE_BY_CANCEL)
```

Comment:
are we sure that this does not have side-effects on other spells? just wondering

Reply:
yes, I'm sure, because this part of the code handles only those 18 different taming spells used for these quests:
https://github.com/azerothcore/azerothcore-wotlk/blob/0b3c5a434ac2cb001ffd59b4ee80f337aeb8189b/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L5000-L5027

Reply:
ok, very well then